client/packages/core/src/store.js

```diff
@@ -106,14 +114,19 @@ export function fromJSON(storeJSON) {
   );
 }
 
+function resetAttrIndexes(store) {
+  store.attrIndexes = createAttrIndexes(store.attrs);
+}
+
 export function createStore(
   attrs,
   triples,
   enableCardinalityInference,
   linkIndex,
 ) {
-  const store = createIndexMap(attrs, triples);
+  const store = createTripleIndexes(attrs, triples);
   store.attrs = attrs;
```

Comment:
At some point, it would be nicer to have an `Attr` abstraction like we do in clojure. 

I skipped it in this PR, as we don't have the really nice protocol abstraction, so we'd have to change callsites